macosx-clang build fails when using latest macos image

**Describe the bug**
`macosx` build fails with a following error:
```
-- Could NOT find MPI_C (missing: MPI_C_WORKS) 
-- Could NOT find MPI_CXX (missing: MPI_CXX_WORKS) 
CMake Error at /usr/local/Cellar/cmake/3.19.6/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:218 (message):
  Could NOT find MPI (missing: MPI_C_FOUND MPI_CXX_FOUND)

      Reason given by package: MPI component 'Fortran' was requested, but language Fortran is not enabled.  

Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.19.6/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:582 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.19.6/share/cmake/Modules/FindMPI.cmake:1722 (find_package_handle_standard_args)
  cmake/load_mpi_package.cmake:6 (find_package)
  cmake/load_packages.cmake:7 (include)
  CMakeLists.txt:61 (include)
```

**To Reproduce**
Occurs at every CI build: https://github.com/DARMA-tasking/vt/actions/workflows/macosx-clang-mpich.yml.
@nmm0 has also spotted this on `kahuna` using `gcc-10`.

**Expected behavior**
Make sure that `macosx` (and any other environment where this was reported) build is passing.

**Additional context**
> @JacobDomagala  seems that the failing builds are using newer version 20210314.1:
https://github.com/actions/virtual-environments/blob/macOS-10.15/20210314.1/images/macos/macos-10.15-Readme.md
https://github.com/actions/virtual-environments/releases/tag/macOS-10.15%2F20210314.1
Where the last build that is green was using 20210307.1:
https://github.com/actions/virtual-environments/blob/macOS-10.15/20210307.1/images/macos/macos-10.15-Readme.md
https://github.com/actions/virtual-environments/releases/tag/macOS-10.15%2F20210307.1

> @PhilMiller The error is actually not necessarily indicative of what actually went wrong in configuration
When it’s missing support for a language, it adds a note about that to the error output that will result, even if the actual cause of failure was unrelated
**The first line of error output in that screenshot is more indicative - there was actually a problem with mpicc and mpicxx, because it ends up without the C and CXX found variables set**

--- a/.github/workflows/macosx-clang-mpich.yml
+++ b/.github/workflows/macosx-clang-mpich.yml
@@ -29,12 +29,17 @@ jobs:
 
     steps:
     - uses: actions/checkout@v2
-    - uses: actions/cache@v1
+    - uses: actions/cache@v2
       with:
         path: ~/.ccache
         key: ${{ runner.os }}-macosx-clang-8-ccache-${{ secrets.GH_ACTIONS_CACHE_VERSION }}-${{ hashFiles('**/*') }}
         restore-keys: |
           ${{ runner.os }}-macosx-clang-8-ccache-${{ secrets.GH_ACTIONS_CACHE_VERSION }}
+    - uses: actions/cache@v2
+      id: mpich-cache
+      with:
+        path: /path/to/mpich
+        key: ${{ runner.os }}-macosx-clang-8-mpich-${{ hashFiles('**/.github/workflows/macosx-clang-mpich.yml') }}
     - name: Install Dependencies
       shell: bash
       run: brew bundle --file=ci/Brewfile
@@ -43,7 +48,8 @@ jobs:
       run: echo "127.0.0.1 $(hostname)" | sudo tee -a /etc/hosts
     - name: Build mpich
       shell: bash
-      run: ./ci/deps/mpich.sh 3.3.2 -j4
+      run: |
+        ci/deps/mpich.sh 3.3.2 -j4
     - name: Build
       shell: bash
       run: ci/build_cpp.sh $(pwd) $(pwd)/build

--- a/.github/workflows/macosx-clang-mpich.yml
+++ b/.github/workflows/macosx-clang-mpich.yml
@@ -41,6 +41,9 @@ jobs:
     - name: Modify hosts file
       shell: bash
       run: echo "127.0.0.1 $(hostname)" | sudo tee -a /etc/hosts
+    - name: Build mpich
+      shell: bash
+      run: ./ci/deps/mpich.sh 3.3.2 -j4
     - name: Build
       shell: bash
       run: ci/build_cpp.sh $(pwd) $(pwd)/build

--- a/.github/workflows/macosx-clang-mpich.yml
+++ b/.github/workflows/macosx-clang-mpich.yml
@@ -38,8 +38,8 @@ jobs:
     - uses: actions/cache@v2
       id: mpich-cache
       with:
-        path: /path/to/mpich
-        key: ${{ runner.os }}-macosx-clang-8-mpich-${{ hashFiles('**/.github/workflows/macosx-clang-mpich.yml') }}
+        path: ~/.mpich
+        key: ${{ runner.os }}-macosx-clang-8-mpich-${{ hashFiles('**/.github/workflows/macosx-clang-mpich.yml', '**/ci/deps/mpich.sh') }}
     - name: Install Dependencies
       shell: bash
       run: brew bundle --file=ci/Brewfile
@@ -47,13 +47,15 @@ jobs:
       shell: bash
       run: echo "127.0.0.1 $(hostname)" | sudo tee -a /etc/hosts
     - name: Build mpich
+      if: steps.mpich-cache.outputs.cache-hit != 'true'
       shell: bash
       run: |
-        ci/deps/mpich.sh 3.3.2 -j4
+        ci/deps/mpich.sh 3.3.2 -j4 $(grealpath ~/.mpich)
     - name: Build
       shell: bash
-      run: ci/build_cpp.sh $(pwd) $(pwd)/build
+      run: |
+        PATH=~/.mpich/bin:$PATH ci/build_cpp.sh $(pwd) $(pwd)/build
     - name: Test
       shell: bash
       run: |
-        ci/test_cpp.sh $(pwd) $(pwd)/build
+        PATH=~/.mpich/bin:$PATH ci/test_cpp.sh $(pwd) $(pwd)/build

--- a/ci/Brewfile
+++ b/ci/Brewfile
@@ -1,2 +1,1 @@
 brew "ccache"
-brew "mpich"

--- a/ci/Brewfile
+++ b/ci/Brewfile
@@ -1,1 +1,2 @@
 brew "ccache"
+brew "coreutils"

--- a/ci/deps/mpich.sh
+++ b/ci/deps/mpich.sh
@@ -4,7 +4,7 @@ set -exo pipefail
 
 if test $# -lt 2
 then
-    echo "usage: ./$0 <mpich-version> <make_flags>"
+    echo "usage: ./$0 <mpich-version> <make_flags> <installation_prefix>"
     exit 1
 fi
 
@@ -12,6 +12,7 @@ mpich_version=$1
 mpich_name="mpich-${mpich_version}"
 mpich_tar_name="${mpich_name}.tar.gz"
 make_flags="$2"
+installation_prefix="${3-}"
 
 echo "${mpich_version}"
 echo "${mpich_name}"
@@ -30,7 +31,8 @@ cd ${mpich_name}
     --enable-fortran=no \
     --enable-fast=all \
     --enable-g=none \
-    --enable-timing=none
+    --enable-timing=none \
+    ${installation_prefix:+ --prefix"=${installation_prefix}"}
 make ${make_flags}
 make install
 cd -

--- a/src/vt/context/context.cc
+++ b/src/vt/context/context.cc
@@ -55,7 +55,10 @@ struct RunnableNew {};
 #endif
 
 #if vt_check_enabled(trace_enabled)
-# include "vt/trace/trace_common.h"
+#include "vt/trace/trace_common.h"
+#if !vt_check_enabled(trace_only)
+#include "vt/context/runnable_context/trace.h"
+#endif
 #endif
 
 #include <string>

